source/fe/mapping_q_generic.cc

```diff
@@ -3737,9 +3770,10 @@ add_quad_support_points(const Triangulation<3,3>::cell_iterator &cell,
                ExcInternalError());
 #endif
 
-      // if face at boundary, then ask boundary object to return intermediate
-      // points on it
-      if (face->at_boundary())
+      // if face at boundary or if it belongs to a manifold, then ask the
+      // boundary/manifold object to return intermediate points on it
+      if (face->at_boundary() ||
+          dynamic_cast<const Boundary<3,3> *>(&face->get_manifold()) == NULL)
```

Comment:
It should be ok to remove the if, and just do `get_intermediate_points_on_object(face->get_manifold(), line_support_points, face, quad_points)` irrespective of being on boundary or not.